sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -35,7 +35,8 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
   }
 
   test("as tuple") {
-    val data = Seq(("a", 1), ("b", 2)).toDF("a", "b")
+    val data = Seq(("a", 1), ("b", 2)).toDF("a", "c").select($"a", $"c".as("b"))
```

Comment:
Oh, i was live coding a demo at Spark summit and accidentally committed the change.  I'll revert.